metadata/src/main/java/org/apache/kafka/controller/FeatureControlManager.java

```diff
@@ -69,7 +69,12 @@
             results.put(entry.getKey(), updateFeature(entry.getKey(), entry.getValue(),
                 downgradeables.contains(entry.getKey()), brokerFeatures, records));
         }
-        return new ControllerResult<>(records, results);
+
+        if (records.isEmpty()) {
```

Comment:
Any reason why we are calling the non-atomic method for an empty record set? Seems like we don't bother doing this in the other control manager classes. Can we just do `ControllerResult.atomicOf(records, results)` regardless of empty records?

Reply:
I don't have a good reason. I felt it was consistent that an empty set of records was by definition not atomic. I removed the conditional and always create an atomic `ControllerResult`.